File: java/mjehuric.java
import java.util.Scanner;

public class mjehuric {
	static int[] p = new int[5];
	static int index;

	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		for (int i = 0; i < p.length; i++)
			p[i] = scan.nextInt();
		while (!(p[0] == 1 && p[1] == 2 && p[2] == 3 && p[3] == 4 && p[4] == 5)) {
			if (index == 4)
				index = 0;
			int nextIndex = index + 1;
			if (p[index] > p[nextIndex]) {
				int tmp = p[index];
				p[index] = p[nextIndex];
				p[nextIndex] = tmp;
				for (int i : p)
					System.out.print(i + " ");
				System.out.println();
			}
			index = nextIndex;
		}
		scan.close();
	}
}

File: ts/src/mjehuric.ts
import readline = require('readline')

type Mjehuric = [number, number, number, number, number]

function isSorted (mjehuric: Mjehuric): boolean {
  const [a, b, c, d, e] = mjehuric
  return a === 1 && b === 2 && c === 3 && d === 4 && e === 5
}

function printSort (mjehuric: Mjehuric): void {
  let index = 0
  while (!isSorted(mjehuric)) {
    if ((mjehuric[index] ?? 0) > (mjehuric[index + 1] ?? 0)) {
      [mjehuric[index], mjehuric[index + 1]] = [
        mjehuric[index + 1] ?? 0,
        mjehuric[index] ?? 0
      ]
      console.log(mjehuric.join(' '))
    }
    index = ++index % 4
  }
}

readline
  .createInterface({
    input: process.stdin,
    output: process.stdout
  })
  .on('line', (line: string) => {
    const [a, b, c, d, e] = line.split(' ').map(Number)
    const pieces: Mjehuric = [a ?? 0, b ?? 0, c ?? 0, d ?? 0, e ?? 0]
    printSort(pieces)
  })
  .on('close', () => {
    process.exit(0)
  })
